# 07 · Promotion simulation
> All promotion results are **scenario-based simulations under stated assumptions**. Online Retail II has no promotion treatment/control, so nothing here is a causal uplift, ROI or observed revenue lift.

EIM = δ·V·m − (p+δ)·V·d − c (SPEC §9).

In [1]:
import os, pathlib
os.chdir(pathlib.Path.cwd().parent if pathlib.Path.cwd().name == "notebooks" else pathlib.Path.cwd())
import pandas as pd, numpy as np, matplotlib.pyplot as plt
from IPython.display import Image, display
pd.set_option("display.width", 200); pd.set_option("display.max_columns", 30)
from retail_targeting.config import load_config
cfg = load_config()
T = lambda name: pd.read_csv(f"outputs/tables/{name}")

In [2]:
from retail_targeting.decision.simulation import scenarios_from_config
pd.DataFrame([s.__dict__ for s in scenarios_from_config(cfg)])

,name,discount_rate,incremental_lift,gross_margin,contact_cost,version
0,conservative,0.05,0.02,0.4,0.1,v1
1,base,0.10,0.05,0.4,0.1,v1
2,aggressive,0.20,0.10,0.4,0.1,v1
3,illustrative_breakeven,0.05,0.10,0.4,0.1,v1


In [3]:
T('break_even.csv')

,scenario,value_quantile,value,p_star
0,conservative,0.1,112.058,0.122152
1,conservative,0.5,282.880,0.132930
2,conservative,0.9,660.604,0.136972
3,base,0.1,112.058,0.141076
4,base,0.5,282.880,0.146465
5,base,0.9,660.604,0.148486
6,aggressive,0.1,112.058,0.095538
7,aggressive,0.5,282.880,0.098232
8,aggressive,0.9,660.604,0.099243
9,illustrative_breakeven,0.1,112.058,0.682152


In [4]:
tt = T('customer_targeting_table.csv'); tt.groupby(['scenario','recommended_action']).size().unstack(fill_value=0)

recommended_action,DO_NOT_TARGET,TARGET
scenario,,
aggressive,5518,0
base,5518,0
conservative,5518,0
illustrative_breakeven,4967,551


Under the three planning scenarios the break-even probability p* is 0.10–0.15, far below observed repeat rates, so the model + value rule recommends **no offer** for every eligible customer.